qemu/tests/ioeventfd.py

```diff
@@ -0,0 +1,163 @@
+                "off" should be less than the one with "on".
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _set_ioeventfd_options():
+        """ Set the ioeventfd options. """
+        if params['drive_format'] == 'virtio':
+            params['blk_extra_params_image1'] = ioeventfd
+            dev_id = 'image1'
+        elif params['drive_format'] == 'scsi-hd':
+            params['bus_extra_params_image1'] = ioeventfd
+            dev_id = 'virtio_scsi_pci0'
+        return dev_id
+
+    def _dd_test():
+        """ Execute dd testing inside guest. """
+        logging.info('Doing dd testing inside guest.')
+        dd_cmd = params['dd_cmd']
+        if os_type == 'windows':
+            dd_cmd = utils_misc.set_winutils_letter(session, dd_cmd)
+        o = session.cmd(dd_cmd, float(params['dd_timeout']))
+        logging.debug(o)
+
+    def _iozone_test(session):
+        """ Execute iozone testing inside guest. """
+        logging.info('Doing iozone inside guest.')
+        if os_type == 'windows':
+            session = utils_test.qemu.windrv_check_running_verifier(
+                session, vm, test, params["driver_name"])
+        try:
+            iozone = generate_instance(params, session, 'iozone')
+            iozone.run(params['iozone_options'], float(params['iozone_timeout']))
+        finally:
+            iozone.clean()
+            return session
+
+    def _check_property(vm, ioeventfd_opt):
+        """
+        Check the value of ioeventfd by sending "info qtree" in QMP.
+        """
+        ioevent_qtree_val = 'true' if 'on' in ioeventfd_opt else 'false'
+        logging.info('Execute info qtree in QMP monitor.')
+        qtree = qemu_qtree.QtreeContainer()
+        qtree.parse_info_qtree(vm.monitor.info('qtree'))
+        for node in qtree.get_nodes():
+            if isinstance(node, qemu_qtree.QtreeDev) and (
+                    node.qtree.get('id', None) == dev_id):
+                if node.qtree.get('ioeventfd', None) is None:
+                    test.fail('The qtree device %s has no property ioeventfd.'
+                              % dev_id)
+                elif node.qtree['ioeventfd'] == ioevent_qtree_val:
+                    logging.info(
+                        'The \"%s\" matches with qtree device \"%s\"(%s).' %
+                        (ioeventfd_opt, dev_id, ioevent_qtree_val))
+                    break
+                else:
+                    test.fail(
+                        'The \"%s\" mismatches with qtree device \"%s\"(%s).' %
+                        (ioeventfd_opt, dev_id, ioevent_qtree_val))
+        else:
+            test.error('No such \"%s\" qtree device.' % dev_id)
+
+    def _get_ioeventfds(ioeventfd_opt):
+        """
+        Get the number of ioeventfds inside host.
+        """
+        logging.info('Check the \"%s\" via /proc/$PID/fd/.' % ioeventfd)
```

Comment:
this event is a case step, better move it front 156 line(function called).

Reply:
As we talked, I'd like to put it inside function.

Reply:
that's ok~